### DISCUSSION TITLE
[issue] swagger组件中，当使用 @Query 装饰参数时，required 未生效

### DISCUSSION THREAD

**Discussion by lengyuxuan** [midwayjs/midway#3883] [Q&A]
### Describe the problem（描述问题）

```typescript
@Get('/xxx')
public async xxxx(@Query() query: GetXxxDto) {
  return 'xxxx';
}

export class GetXxxDto{
  @ApiProperty({
    required: true,
  })
  @Rule(RuleType.string().required())
  corpCode: string;
}
```
![image](https://github.com/midwayjs/midway/assets/23334732/add95241-ac29-4e8b-9459-081a1ce5ee52)

我查看了对应的代码，@midwayjs/swagger@3.15.3 和 @midwayjs/swagger@3.16.1 相关位置的代码有过变动，但是两个版本都是错误的。

在 `swaggerExplorer.ts` 中的 `parseClzz` 
![image](https://github.com/midwayjs/midway/assets/23334732/ae8484c2-9bdc-4070-9c3a-82079adbdc62)

但是在同文件 `generateRouteMethod` 中却没有使用收集到数组，似乎忘记了对 required 的操作：
![image](https://github.com/midwayjs/midway/assets/23334732/e42bce7c-eab5-4b66-b5b3-e2d1c37c534c)
做出如下修正后，表现正常：
![image](https://github.com/midwayjs/midway/assets/23334732/5017b847-a87a-4b35-9ce2-c34ab52bea4b)

但由于工作关系，我没有足够的时间大规模阅读代码，不知此变动是否对项目有其他影响，辛苦工作人员对此进行评估。
还望尽快修复此问题。

### Midway Versions（Midway 版本）

@midwayjs/core@3.12.0
@midwayjs/swagger@3.16.1

**Comment by czy88840616** [comment]
v3.16.2
